src/providers/consensus/client.py

```diff
@@ -38,6 +40,20 @@ def get_block_root(self, state_id: Union[str, SlotNumber, BlockRoot]) -> BlockRo
         data, _ = self._get(self.API_GET_BLOCK_ROOT.format(state_id))
         return BlockRootResponse(**data)
 
+    def get_block_header(self, state_id: Union[str, SlotNumber, BlockRoot]) -> BlockHeaderFullResponse:
+        """
+        Spec: https://ethereum.github.io/beacon-APIs/#/Beacon/getBlockRoot
+
+        No cache because this method is using to get finalized and head block, and they could not be cached by args.
```

Comment:
As I see we don't use this method with finalized and head params, so can use cache here

Reply:
fixed